Runtime/Scripts/DebugUtils/BillboardText.cs

```diff
@@ -0,0 +1,47 @@
+using TMPro;
+using UnityEngine;
+
+public class BillboardText : MonoBehaviour
+{
+
+    private Camera _mainCamera;
+    private TextMeshProUGUI _text;
+    private string _content;
+    private float _yOffset;
+    
+    void Awake()
+    {
+        _mainCamera = Camera.main;
+        _text = GetComponentInChildren<TextMeshProUGUI>();
+        if (_content != null)
+        {
+            SetText(_content);
+            SetYOffset(_yOffset);
+        }
+    }
+    
+    void LateUpdate()
+    {
+        Vector3 newRotation = _mainCamera.transform.eulerAngles;
+        newRotation.x = 0;
+        newRotation.z = 0;
+        transform.eulerAngles = newRotation;

```

Comment:
This is still better, but wrong... I liked how they have it in BossRoom, but it seems like what they do is get the position of the character on the screen and then draw the UI onto a canvas that covers the entire screen, rather than actually have a canvas on the player. Maybe this is good enough for now, but I notice sometimes it can be janky.

Reply:
have a look at `LookAtCameraInEditor` in our SDK

I apply this to the `NameTag` prefab in our tools so that the textmesh pro gameObjects floating above the characters are always facing to a point 'infinitely' behind the camera (not really infinite.. but a long long way).

There are other solutions to the problem, but you should be able to do something nearly identical here and/or move & rename & adjust that current script to work in editor and non-editor mode.

Reply:
Did you try the `LookAt` method previously? What's the issue with the billboarding?
Ex:

```
if (_mainCamera != null)
{
   transform.LookAt(transform.position + _mainCamera.transform.rotation * Vector3.forward,
       _mainCamera.transform.rotation * Vector3.up);
}
```